server/utils/__tests__/is-protected-entity.spec.js

```diff
@@ -115,11 +115,107 @@ describe('isProtectedEntity', () => {
+  });
+
+  it('should return `false` if entity attribute does not match the `lte` rule', () => {
+    const entity = { attribute: 1 };
+    const rules = [['attribute', 'lte', 0]];
+    const result = isProtectedEntity(entity, rules);
+
+    expect(result).toBe(false);
+  });
+
+  it('should return `true` if entity attribute matches the `gt` rule', () => {
+    const entity = { attribute: 1 };
+    const rules = [['attribute', 'gt', 0]];
+    const result = isProtectedEntity(entity, rules);
+
+    expect(result).toBe(true);
+  });
+
+  it('should return `false` if entity attribute does not match the `gt` rule', () => {
+    const entity = { attribute: 1 };
+    expect(isProtectedEntity(entity, [['attribute', 'gt', 1]])).toBe(false);
+    expect(isProtectedEntity(entity, [['attribute', 'gt', 2]])).toBe(false);
+  });
+
+  it('should return `true` if entity attribute matches the `gte` rule', () => {
+    const entity = { attribute: 1 };
+
+    expect(isProtectedEntity(entity, [['attribute', 'gte', 1]])).toBe(true);
+    expect(isProtectedEntity(entity, [['attribute', 'gte', 0]])).toBe(true);
+  });
+
+  it('should return `false` if entity attribute does not match the `gte` rule', () => {
+    const entity = { attribute: 1 };
+    const rules = [['attribute', 'gte', 2]];
+    const result = isProtectedEntity(entity, rules);
+
+    expect(result).toBe(false);
+  });
+
+  it('should return `true` if entity attribute matches the `between` rule', () => {
+    const entity = { attribute: 1 };
+
+    expect(isProtectedEntity(entity, [['attribute', 'between', [0, 2]]])).toBe(true);
+    expect(isProtectedEntity(entity, [['attribute', 'between', [0, 1]]])).toBe(true);
+    expect(isProtectedEntity(entity, [['attribute', 'between', [1, 2]]])).toBe(true);
+  });
+
+  it('should return `false` if entity attribute does not match the `between` rule', () => {
+    const entity = { attribute: 1 };
+
+    expect(isProtectedEntity(entity, [['attribute', 'between', [-2, 0]]])).toBe(false);
+    expect(isProtectedEntity(entity, [['attribute', 'between', [2, 4]]])).toBe(false);
+  });
+
   it('should return `false` if the provided `comparator` is unknown', () => {
     const entity = { type: 'user' };
     const rules = [['name', 'unknown_comparator', 'user']];
     const result = isProtectedEntity(entity, rules);
 
     expect(result).toBe(false);
   });
+
+  it('should return `false` if all the provided rules are falsy', () => {
+    const entity = { attribute: 1 };
+    const rules = [
+      ['attribute', 'is', 0],
+      ['attribute', 'is', 2],
+    ];
+    const result = isProtectedEntity(entity, rules);
+
+    expect(result).toBe(false);
+  });
+
+  it('should return `true` if at least one of the provided rules is truly', () => {
+    const entity = { attribute: 1 };
+    const rules = [
+      ['attribute', 'is', 0],
+      ['attribute', 'is', 1],
+    ];
+    const result = isProtectedEntity(entity, rules);
+
+    expect(result).toBe(false);
```

Comment:
@BorysShulyak This should `expect` a `true` value, not `false`.

Reply:
Done.